src/sync/atomic/mod.rs

```diff
@@ -172,18 +190,46 @@ impl<T: Copy + Eq> Atomic<T> {
         maybe_warn_about_ordering(fetch_order);
 
         thread::switch();
+        self.exhale_clock();
         let current = *self.inner.borrow();
         let ret = if let Some(new) = f(current) {
             *self.inner.borrow_mut() = new;
             Ok(current)
         } else {
             Err(current)
         };
+        self.inhale_clock();
         thread::switch();
         ret
     }
 
     unsafe fn raw_load(&self) -> T {
+        self.exhale_clock();
```

Comment:
I don't think we want to modify the clock here; the idea is that `raw_load` doesn't affect synchronization operations at all (so we can insert uses of it when debugging it to inspect an atomic's value without perturbing a replay). I realize now that I put a comment to that effect on `AtomicBool::raw_load` and friends but not on this method.

Reply:
Ah, got it.  Removed.